test/common/config/subscription_factory_impl_test.cc

```diff
@@ -35,9 +35,10 @@ class SubscriptionFactoryTest : public testing::Test {
 
   std::unique_ptr<Subscription>
   subscriptionFromConfigSource(const envoy::api::v2::core::ConfigSource& config) {
-    return SubscriptionFactory::subscriptionFromConfigSource(
-        config, local_info_, dispatcher_, cm_, random_, stats_store_,
-        Config::TypeUrl::get().ClusterLoadAssignment, validation_visitor_, *api_, callbacks_);
+    return SubscriptionFactoryImpl(local_info_, dispatcher_, cm_, random_, validation_visitor_,
+                                   *api_)
+        .subscriptionFromConfigSource(config, Config::TypeUrl::get().ClusterLoadAssignment,
```

Comment:
Since the rvalue SubscriptionFactoryImpl instance is dying, it's hard to determine whether the `Subscription` is healthy. Maybe it's correct in certain conditions...

Reply:
I understand the point on the rvalue going away, but not why this is an issue. Can you elaborate?

Reply:
I looked into `SubscriptionFactoryImpl` in `source/common/config/subscription_factory_impl.h`. `SubscriptionFactoryImpl` doesn't own anything: each member is reference. So it's good. Sorry for the false alarm